README.md

```diff
@@ -9,18 +9,140 @@
+
+### `GOOGLE_PRIVATE_KEY`
+
+This is the private key associated with the Google Cloud Platform Service account created above. In order to retrieve the correct value, see [Creating and Managing Service Account Keys in the Google Cloud Platform documentation](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) or continue the step-by-step guide from above:
+
+6. Make sure you've followed all steps to get the value of `GOOGLE_CLIENT_EMAIL`
+7. Copy the value of the `private_key` property in the JSON file you've downloaded
+
+**Note:** The private key and the value of the `GOOGLE_PRIVATE_KEY` environment variable should be considered private and should never be checked in to source control.
+
+The private key is a multi-line value.
+
+**Note:** Private keys created using an API typically have a short expiration time and need to be rotated in regular intervals. Even for private keys that have been created manually, regular rotation is a best practice.
+
+### `GOOGLE_PROJECT_ID`
+
+This is the Google Cloud Platform project ID. It looks like `project-12345678` and you will find it in lots of places in the Google Cloud Platform Console UI. In addition, you can just take the value of the `project_id` property in your downloaded key JSON file.
+
+### `HLX_FASTLY_NAMESPACE`
+
+This property is only required for testing and development. It is the service config ID that you can retrieve from Fastly.
+
+For testing, it is a good idea to use a separate, non-production service config, as the tests not only perform frequent updates, but they also rotate the private keys of the created Google Cloud Platform service accounts. As the tests don't activate the service config, this will lead to an invalid logging configuration in a short time.
+
+### `HLX_FASTLY_AUTH`
+
+This property is only required for testing and development. It is an API token for the Fastly API. Follow the [instructions in the Fastly documentation](https://docs.fastly.com/guides/account-management-and-security/using-api-tokens) to create a token.
+
+The token needs to have `global`, i.e. write access to your service config.
+
+
+**Note:** The API token and the value of the `HLX_FASTLY_AUTH` environment variable should be considered private and should never be checked in to source control.
 For more, see the [API documentation](docs/API.md).
 
 ## Development
 
-### Deploying Helix Run Query
+You need node>=8.0.0 and npm>=5.4.0. Follow the typical npm install, npm test workflow.
+
+Contributions are highly welcome.
+
+### Queries
+
+    Query files live in the src/queries directory. They are static resources; that are loaded into run-query and
+    then sent to Bigquery for actual execution. It is up to the developer to ensure their query is correct; this 
+    can be done by using the Bigquery console. 
+
+    Once a query file is complete and correct, you may add it as a static resource; so that it won't be excluded during openwhisk deployment. 
```

Comment:
```suggestion
    Once a query file is complete and correct, you may add it as a static resource; so that it won't be excluded during OpenWhisk deployment. 
```